cli/lsp/documents.rs

```diff
@@ -606,46 +606,81 @@ fn recurse_dependents(
 }
 
 #[derive(Debug, Default)]
-struct DocumentsInner {
-  /// The DENO_DIR that the documents looks for non-file based modules.
+struct SpecifierResolver {
```

Comment:
Extracted this out to isolate the `redirects` mutation behind a mutex because it is used by methods on `Document` that don't require mutation now such as `get`.